src/App.tsx

```diff
@@ -1,14 +1,59 @@
 /* eslint-disable max-len */
-import React from 'react';
+import React, { useState, useEffect } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { Todo } from './types/Todo';
+import { getTodos, getUser } from './api';
+import { User } from './types/User';
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [loadingTodo, setLoadingTodo] = useState(true);
+  const [loadingUser, setLoadingUser] = useState(true);
+
+  const [selectedTodo, setSelectedTodo] = useState<Todo | null>(null);
+  const [selectedUser, setSelectedUser] = useState<User | null>(null);
+
+  const [selectedOption, setSelectedOption] = useState<string>('all');
+  const [query, setQuery] = useState<string>('');
+
+  function handleSelectedOption(event: React.ChangeEvent<HTMLSelectElement>) {
+    setSelectedOption(event.target.value as string);
```

Comment:
The type casting `as string` is unnecessary here because `event.target.value` is already a string. You can directly assign `event.target.value` to `setSelectedOption`.